### PR TITLE
BP-634 Update all .NET nuget packages in the building blocks

### PR DESCRIPTION
**Pull request by avanwaas**
https://enigmatry.atlassian.net/browse/BP-634 Update all .NET nuget packages in the building blocks

- Disable warning EnableGenerateDocumentationFile introduced in VS 2022 17.7.5 (https://github.com/dotnet/roslyn/issues/41640)
- Remove unused packages from Directory.Packages.props
- Remove unused dependency FluentValidation.AspNetCore
- Remove deprecated dependency Microsoft.Rest.ClientRuntime
- Update dependencies:
  * AspNetCore.HealthChecks.System to 7.0.0
  * AutoMapper to 12.0.1
  * Azure.Identity to 1.10.3
  * Azure.Storage.Blobs to 12.18.0
  * CachingFramework.Redis to 15.0.2
  * CachingFramework.Redis.NewtonsoftJson to 15.0.2
  * coverlet.collector to 6.0.0
  * coverlet.msbuild to 6.0.0
  * CsvHelper to 30.0.1
  * EmailValidation to 1.0.10
  * FakeItEasy to 7.4.0
  * FakeItEasy.Analyzer.CSharp to 6.1.1
  * FluentAssertions to 6.12.0
  * FluentValidation to 11.8.0
  * JetBrains.Annotations to 2023.2.0
  * MailKit to 4.2.0
  * MediatR to 12.1.1
  * Microsoft.AspNet.WebApi.Client to 6.0.0
  * Microsoft.AspNetCore.Mvc.NewtonsoftJson to 7.0.13
  * Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation to 7.0.13
  * Microsoft.AspNetCore.Mvc.Testing to 7.0.13
  * Microsoft.EntityFrameworkCore to 7.0.13
  * Microsoft.EntityFrameworkCore.SqlServer to 7.0.13
  * Microsoft.Extensions.Caching.Abstractions to 7.0.0
  * Microsoft.Extensions.Configuration to 7.0.0
  * Microsoft.Extensions.Configuration.Abstractions to 7.0.0
  * Microsoft.Extensions.Configuration.Binder to 7.0.4
  * Microsoft.Extensions.Configuration.EnvironmentVariables to 7.0.0
  * Microsoft.Extensions.Configuration.UserSecrets to 7.0.0
  * Microsoft.Extensions.DependencyInjection to 7.0.0
  * Microsoft.Extensions.FileProviders.Physical to 7.0.0
  * Microsoft.Extensions.Hosting.Abstractions to 7.0.0
  * Microsoft.Extensions.Logging to 7.0.0
  * Microsoft.Extensions.Logging.Abstractions to 7.0.1
  * Microsoft.Extensions.Logging.Console to 7.0.0
  * Microsoft.Extensions.Options to 7.0.1
  * Microsoft.Extensions.Options.ConfigurationExtensions to 7.0.0
  * Microsoft.NET.Test.Sdk to 17.7.2
  * MimeKit to 4.2.0
  * Newtonsoft.Json to 13.0.3
  * NSwag.AspNetCore to 13.20.0
  * NUnit.Analyzers to 3.8.0
  * NUnit3TestAdapter to 4.5.0
  * Polly to 8.0.0
  * Quartz to 3.7.0
  * Quartz.Extensions.DependencyInjection to 3.7.0
  * Quartz.Extensions.Hosting to 3.7.0
  * Serilog to 3.0.1
  * System.Linq.Dynamic.Core to 1.3.5
  * System.Net.Http.Json to 7.0.1
  * Verify.NUnit to 22.1.4
- Fix warnings

### CODE REVIEW

File: Enigmatry.Entry.AzureSearch.Tests/Searching/SearchServiceFixtureBase.cs [hunk 1]
```diff
@@ -25,6 +25,9 @@ public async Task OneTimeSetup()
         WaitIndexToBeUpdated();
     }
 
+    [OneTimeTearDown]
+    public void OneTimeTearDown() => _services.Dispose();
```
Comment (zsrdjan) [inline Enigmatry.Entry.AzureSearch.Tests/Searching/SearchServiceFixtureBase.cs:29]: Why is this needed now. and does it conflict with the [TearDown] in a particular test? Will you dispose twice?

**Review by zsrdjan — APPROVED** [review]

File: Enigmatry.Entry.AzureSearch.Tests/SearchIndexManagerFixture.cs [hunk 2]
```diff
@@ -20,6 +20,9 @@ public void Setup()
         _indexManager = _services.GetRequiredService<ISearchIndexManager<TestDocument>>();
     }
 
+    [TearDown]
+    public void TearDown() => _services.Dispose();
```
[the remark this answers is not in the document]
Comment (avanwaas) [reply Enigmatry.Entry.AzureSearch.Tests/SearchIndexManagerFixture.cs:24]: warning was generated by NUnit.Analyzers

File: Enigmatry.Entry.AzureSearch.Tests/Searching/SearchServiceFixtureBase.cs (hunk 1 above)
Comment (avanwaas) [reply Enigmatry.Entry.AzureSearch.Tests/Searching/SearchServiceFixtureBase.cs:29]: Warning was reported by NUnit.Analyzers that it wasn't disposed. _services is initialized once in OneTimeSetup, and now disposed in OneTimeTearDown.
